SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DynamicSubobjectsTest/SpatialDynamicComponentsFastReadditionTest.h

```diff
@@ -0,0 +1,46 @@
+﻿#pragma once
+
+#include "Components/ActorComponent.h"
+#include "GameFramework/Actor.h"
+
+#include "SpatialFunctionalTest.h"
+
+#include "SpatialDynamicComponentsFastReadditionTest.generated.h"
+
+UCLASS()
+class ADynamicComponentsTestActor : public AActor
+{
+public:
+	GENERATED_BODY()
+
+	ADynamicComponentsTestActor();
+};
+
+UCLASS()
+class USelfRecreatingDynamicComponent : public UActorComponent
+{
+public:
+	GENERATED_BODY()
+
+	USelfRecreatingDynamicComponent();
+
+	virtual void PreReplication(IRepChangedPropertyTracker& ChangedPropertyTracker) override;
+
+private:
+	bool bNeedsToRecreateItself = true;
```

Comment:
I think I would prefer this in the constructor, just because I was looking for it in the cpp, but not sure what our preferences in general for these things are.

Reply:
had same question on a pr a while ago, we should probably make some guideline one way or the other

Reply:
Here's the guidance from Epic, whose coding standard we're using: https://docs.unrealengine.com/4.26/en-US/ProductionPipelines/DevelopmentSetup/CodingStandard/#defaultmemberinitializers
Overall, I'm more in favor of initializing members in headers, although this certainly goes against the norm in current GDK code.